ambry-router/src/test/java/com.github.ambry.router/CloudRouterTest.java

```diff
@@ -0,0 +1,277 @@
+import com.github.ambry.config.RouterConfig;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.messageformat.MessageFormatRecord;
+import com.github.ambry.network.LocalNetworkClientFactory;
+import com.github.ambry.network.LocalRequestResponseChannel;
+import com.github.ambry.network.NetworkClientFactory;
+import com.github.ambry.network.NetworkMetrics;
+import com.github.ambry.notification.NotificationSystem;
+import com.github.ambry.protocol.RequestHandlerPool;
+import com.github.ambry.utils.SystemTime;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Properties;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Class to test the {@link NonBlockingRouter} returned by {@link CloudRouterFactory}.
+ */
+@RunWith(Parameterized.class)
+public class CloudRouterTest extends NonBlockingRouterTest {
+  private static final int SUCCESS_TARGET = 1;
+  private static final int REQUEST_PARALLELISM = 1;
+  private static final Logger logger = LoggerFactory.getLogger(CloudRouterTest.class);
+
+  /**
+   * Running for both regular and encrypted blobs, and versions 2 and 3 of MetadataContent
+   * @return an array with all four different choices
+   */
+  @Parameterized.Parameters
+  public static List<Object[]> data() {
+    return Arrays.asList(new Object[][]{{false, MessageFormatRecord.Metadata_Content_Version_V2},
+        {false, MessageFormatRecord.Metadata_Content_Version_V3},
+        {true, MessageFormatRecord.Metadata_Content_Version_V2},
+        {true, MessageFormatRecord.Metadata_Content_Version_V3}});
+  }
+
+  /**
+   * Initialize parameters common to all tests.
+   * @param testEncryption {@code true} to test with encryption enabled. {@code false} otherwise
+   * @throws Exception
+   */
+  public CloudRouterTest(boolean testEncryption, int metadataContentVersion) throws Exception {
+    super(testEncryption, metadataContentVersion);
+  }
+
+  @After
+  public void after() {
+    // Any single test failure should not impact others.
+    if (router != null && router.isOpen()) {
+      router.close();
+    }
+    Assert.assertEquals("Current operations count should be 0", 0, NonBlockingRouter.currentOperationsCount.get());
+  }
+
+  /**
+   * Constructs and returns a VerifiableProperties instance with the defaults required for instantiating
+   * the {@link NonBlockingRouter}.
+   * @return the created VerifiableProperties instance.
+   */
+  @Override
+  protected Properties getNonBlockingRouterProperties(String routerDataCenter) {
+    Properties properties = super.getNonBlockingRouterProperties(routerDataCenter);
+    // Set parallelism and success quorum to 1 for all operations
+    properties.setProperty("router.put.request.parallelism", Integer.toString(REQUEST_PARALLELISM));
```

Comment:
adding a note that we will eventually have to consider changing how these configs work when we add support for addressing both LocalNetworkClient and SocketNetworkClient. This will require more significant changes to OperationTracker